sync/src/relayer/mod.rs

```diff
@@ -419,20 +420,61 @@ impl<CS: ChainStore + 'static> Relayer<CS> {
                     peer,
                 );
             }
-            for tx_hash in tx_hashes {
-                let fbb = &mut FlatBufferBuilder::new();
-                let message = RelayMessage::build_get_transaction(fbb, &tx_hash);
-                fbb.finish(message, None);
-                let data = fbb.finished_data().into();
-                if let Err(err) = nc.send_message_to(*peer, data) {
-                    debug_target!(
-                        crate::LOG_TARGET_RELAY,
-                        "relayer send Transaction error: {:?}",
-                        err,
-                    );
-                    break;
+
+            let fbb = &mut FlatBufferBuilder::new();
+            let message = RelayMessage::build_get_transactions(fbb, &tx_hashes);
+            fbb.finish(message, None);
+            let data = fbb.finished_data().into();
+            if let Err(err) = nc.send_message_to(*peer, data) {
+                debug_target!(
+                    crate::LOG_TARGET_RELAY,
+                    "relayer send Transaction error: {:?}",
+                    err,
+                );
+                break;
+            }
+        }
+    }
+
+    // Send bulk of tx hashes to selected peers
+    pub fn send_bulk_of_tx_hashes(&self, nc: &CKBProtocolContext) {
+        let mut selected: FnvHashMap<PeerIndex, FnvHashSet<H256>> = FnvHashMap::default();
+        {
+            let peer_tx_hashes = self.shared.take_tx_hashes();
+            let mut known_txs = self.shared.known_txs();
+
+            for (peer_index, tx_hashes) in peer_tx_hashes.into_iter() {
+                for tx_hash in tx_hashes {
+                    let peers: Vec<PeerIndex> = nc
+                        .connected_peers()
+                        .into_iter()
+                        .filter(|target_peer| {
+                            known_txs.insert(*target_peer, tx_hash.clone())
+                                && (peer_index != *target_peer)
+                        })
+                        .take(MAX_RELAY_PEERS)
+                        .collect();
```

Comment:
no need to allocate a new vec here, we can use `for_each` after `take` directly

Reply:
updated, thanks.